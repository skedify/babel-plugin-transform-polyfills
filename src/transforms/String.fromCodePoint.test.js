import compile from '../../test/utils/compile'

describe('String.fromCodePoint', () => {
  it('should replace String.fromCodePoint with a polyfill', () => {
    const result = compile(
      `
      const c = String.fromCodePoint(1, 2)
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = String.fromCodePoint(1)
      const b = String.fromCodePoint(2)
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(String.fromCodePoint(1, 2))
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when using apply', () => {
    const result = compile(
      `
      const c = String.fromCodePoint.apply(null, [1, 2, 3])
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when using call', () => {
    const result = compile(
      `
      const c = String.fromCodePoint.call(null, 1, 2)
    `
    )

    expect(result).toMatchSnapshot()
  })
})
