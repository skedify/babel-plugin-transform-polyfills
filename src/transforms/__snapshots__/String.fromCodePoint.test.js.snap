// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String.fromCodePoint should not import polyfill more than once 1`] = `
"import _String_fromCodePoint from \\"babel-plugin-transform-polyfills/polyfills/String/fromCodePoint\\";

const e = _String_fromCodePoint(1);

const b = _String_fromCodePoint(2);"
`;

exports[`String.fromCodePoint should replace String.fromCodePoint with a polyfill 1`] = `
"import _String_fromCodePoint from \\"babel-plugin-transform-polyfills/polyfills/String/fromCodePoint\\";

const c = _String_fromCodePoint(1, 2);"
`;

exports[`String.fromCodePoint should work when part of a call chain 1`] = `
"import _String_fromCodePoint from \\"babel-plugin-transform-polyfills/polyfills/String/fromCodePoint\\";
const c = Object.keys(_String_fromCodePoint(1, 2));"
`;

exports[`String.fromCodePoint should work when using apply 1`] = `
"import _String_fromCodePoint from \\"babel-plugin-transform-polyfills/polyfills/String/fromCodePoint\\";

const c = _String_fromCodePoint.apply(null, [1, 2, 3]);"
`;

exports[`String.fromCodePoint should work when using call 1`] = `
"import _String_fromCodePoint from \\"babel-plugin-transform-polyfills/polyfills/String/fromCodePoint\\";

const c = _String_fromCodePoint.call(null, 1, 2);"
`;
